made the basic webservice to be used

vultouri03 opened the pull request:
Created a basic webservice, this has a post and a get, the get is for testing and the post can be used to send base64 to the server, the response is a premade categories array

vultouri03 committed:
made the basic webservice to be used

Created a basic webservice, this has a post and a get, the get is for testing and the post can be used to send base64 to the server, the response is a premade categories array

ZaidProjectsDev commented:
**Failed** 
node:buffer:1335
    throw lazyDOMException(
          ^
DOMException [InvalidCharacterError]: The string to be decoded is not correctly encoded.
    at atob (node:buffer:1335:11)
    at file:///D:/School/Development/TLE2/TLE-2-receipt-scan-webservice/router.js:24:17
    at Layer.handle [as handle_request] (D:\School\Development\TLE2\TLE-2-receipt-scan-webservice\node_modules\express\lib\router\layer.js:95:5)
    at next (D:\School\Development\TLE2\TLE-2-receipt-scan-webservice\node_modules\express\lib\router\route.js:149:13)
    at Route.dispatch (D:\School\Development\TLE2\TLE-2-receipt-scan-webservice\node_modules\express\lib\router\route.js:119:3)
    at Layer.handle [as handle_request] (D:\School\Development\TLE2\TLE-2-receipt-scan-webservice\node_modules\express\lib\router\layer.js:95:5)
    at D:\School\Development\TLE2\TLE-2-receipt-scan-webservice\node_modules\express\lib\router\index.js:284:15
    at Function.process_params (D:\School\Development\TLE2\TLE-2-receipt-scan-webservice\node_modules\express\lib\router\index.js:346:12)      
    at next (D:\School\Development\TLE2\TLE-2-receipt-scan-webservice\node_modules\express\lib\router\index.js:280:10)
    at Function.handle (D:\School\Development\TLE2\TLE-2-receipt-scan-webservice\node_modules\express\lib\router\index.js:175:3)

What I did : I POST to 8000/classify without sending a param key.

vultouri03 committed:
fixed a bug that allowed the sending of empty data

added an if statement to check if the image data has been filled, if it has not it will send back a default array

vultouri03 commented:
should have been fixed with this last push

ZaidProjectsDev commented:
**Passed**
Works and returns: 
{
    "dairy": 10,
    "meat": 0,
    "drinks": 0,
    "vegetables": 0,
    "wheats": 0,
    "spices": 0,
    "sauces": 0
}

ZaidProjectsDev reviewed (approved):
Works as intended. Sends: 
{
    "dairy": 10,
    "meat": 0,
    "drinks": 0,
    "vegetables": 0,
    "wheats": 0,
    "spices": 0,
    "sauces": 0
}
